src/sentry/integrations/slack/service.py

```diff
@@ -190,7 +216,21 @@ def _get_notification_message_to_send(self, activity: Activity) -> str | None:
 
         Apparently the get_context is a very computation heavy call, so make sure to only call this once.
         """
-        notification_cls = EMAIL_CLASSES_BY_TYPE.get(activity.type, None)
+        try:
+            activity_type: ActivityType = ActivityType(activity.type)
+        except Exception as err:
```

Comment:
is there a more specific error you can catch on?

Reply:
in what cases would this error? could we add a test if so?

Reply:
I want to log regardless of the error type, which is why this is generic. But for what it's worth, if the value is not part of the enum, it will throw a ValueError, which is pretty standard behavior for all enums

![Screenshot 2024-04-17 at 10 29 27 AM](https://github.com/getsentry/sentry/assets/5581484/abcf3654-900d-4e4e-bdcd-76fbd227a532)
![Screenshot 2024-04-17 at 10 29 38 AM](https://github.com/getsentry/sentry/assets/5581484/241856f3-2728-4aea-bf6b-1f1acc13c870)

Reply:
so why not catch on `ValueError` then if that's the only error possible?

Reply:
> so why not catch on ValueError then if that's the only error possible?

We can limit it to that specific error type sure, I don't mind! But I do want to understand the hesitation around why we want to scope it down

Reply:
> A good rule of thumb is to limit use of bare ‘except’ clauses to two cases:
> - If the exception handler will be printing out or logging the traceback; at least the user will be aware that an error has occurred.
>  - If the code needs to do some cleanup work, but then lets the exception propagate upwards with raise. try...finally can be a better way to handle this case.

In our case here, we log, and we exit early and safely upwards out of this function so the parent function can handle the behavior based on it's needs. But, I did change the error type to ValueError!

Reply:
got it thank you!